runtime/queries/rust/highlights.scm

```diff
@@ -154,30 +151,24 @@
   "trait"
   "for"
 
-  "unsafe"
   "default"
-  "macro_rules!"
-
   "async"
 ] @keyword
 
 [
   "struct"
   "enum"
   "union"
-
   "type"
 ] @keyword.storage.type
 
 "let" @keyword.storage
-
 "fn" @keyword.function
+"unsafe" @keyword.special
+"macro_rules!" @function.macro
 
 (mutable_specifier) @keyword.storage.modifier.mut
 
-(reference_type "&" @keyword.storage.modifier.ref)
-(self_parameter "&" @keyword.storage.modifier.ref)
-
```

Comment:
These were intentional: https://github.com/helix-editor/helix/pull/2731#discussion_r893982611

Reply:
Ah that's fair enough, I've put them back. On further investigation it looks like `@keyword.storage.modifier.ref` is specific enough to not clash with other storage modifier keywords if overridden in a theme anyway.